src/stores/FiltersHistoryStore.ts

```diff
@@ -174,9 +166,38 @@ class FiltersHistoryStore {
 		const history = await this.indexedDb.getStoreValues<
 			FiltersHistoryType<EventFilterState | MessageFilterState>
 		>(IndexedDbStores.FILTERS_HISTORY);
-
 		this.filterHistory = history.sort(sortByTimestamp);
+		this.initialized = true;
+	};
+
+	@action
+	private addEventHistoryItem = async (newItem: FiltersHistoryType<EventFilterState>) => {
+		if (this.eventsHistory.some(({ filters }) => isEqual(filters, newItem.filters))) return;
```

Comment:
I think it's better to change this behaviour.
If we find same filter lets update it's timestamp.